popup/search-results.js

```diff
@@ -0,0 +1,547 @@
+/* global handleEvent tryJSONparse getStylesSafe BG */
+'use strict';
+
+/**
+ * Library for interacting with userstyles.org
+ * @returns {Object} Exposed methods representing the search results on userstyles.org
+ */
+function SearchUserstyles() {
+  let totalPages;
+  let currentPage = 1;
+  let exhausted = false;
+
+  return {getCurrentPage, getTotalPages, getCategory, isExhausted, search, fetchStyleJson, fetchStyle};
+
+  /**
+   * @returns {Number} The *Next* page to fetch for styles. Auto-incremented after each search.
+   */
+  function getCurrentPage() {
+    return currentPage;
```

Comment:
While it looks mature and all, but we don't need these getters outside of the closure. We can add them when it's actually needed, if ever.